subprojects/libplacebo.wrap

```diff
@@ -0,0 +1,6 @@
+[wrap-git]
+
+url = https://code.videolan.org/videolan/libplacebo.git
+revision = v6.338.0
```

Comment:
Why hard-code any revision at all? Isn't the whole point of a wrap (vs a submodule) to *not* need to pin specific versions?

Reply:
Hmm seems to be no easy way to fetch the latest tag. A bit unfortunate.

Reply:
If we track HEAD, mpv release would essentially be blocked by libplacebo releases. Advantage is that we can integrate new libplacebo features sooner during the development version, but this makes mpv compatible only with HEAD version and releases would need to be done with that kept in mind.

If we track Releases, mpv is treating libplacebo as any other dependency, possibly with new features added behind #ifdef.

I think 2nd option is more sane and easy to maintain in future. Of course we need manually bump libplacebo version, but this is also good time to clear any possible #ifdefs that has been added and no longer needed.

The main issue that we want to resolve is to be able to track Release version of libplacebo as soon as possible, and not wait for all distros to update. While both submodule and wrap resolves this issue. Still there is an advantage over submodule, because wrap are working nicely with meson integration.

Reply:
> Hmm seems to be no easy way to fetch the latest tag. A bit unfortunate.

You can track the `release` branch.

Reply:
> You can track the release branch, which always points to the latest tag. (Though it may get force-pushed)

We actually need to test this in meson, if it actually works.

What I noticed is that after modifying wrap file meson will display warning that wrap file has changed and subproject may be outdated, but **will not** update it, even on `meson setup --wipe` which is quite annoying, because of far I needed to manually remove directory.

Reply:
I don't think it works, I think having to update this file every libplacebo release is fine. I think tracking HEAD is fine since libplacebo's release cadence is much higher than mpv, and the issue of needing to schedule libplacebo releases with mpv sometimes isn't a big issue.

But if we are tracking the latest tag, then I think changing the revision every once in a while is fine to not deal with what kasper mentioned.

Reply:
I also meant that changing the revision itself doesn't automatically update. It just need to be tested, so we don't introduce somethings that will be PITA to use in the future.

Reply:
Seems like we've agreed to just update it on releases for the time being.